### ISSUE TITLE
Azure Cloud deployments using Terraform/Opentofu possible/planned ?

### ISSUE THREAD

**Issue by AkechiShiro** [aanderse/teraflops#7]
Hi,

I would like to use this to deploy in Azure VMs running NixOS and manage them, do you think it could be possible to achieve this ?

**Comment by aanderse** [comment]
i found out that `opentofu` support isn't great in nixpkgs yet... but it is being worked on, so i'm waiting on that at the moment to avoid having to write workarounds

any cloud provider which has a `terraform` plugin is relatively simple to get working
[existing plugins](https://github.com/aanderse/teraflops/tree/main/nix) are a good example to work from

have you ever used the `terraform` provider for `azure`?

**Comment by AkechiShiro** [comment]
Not yet, but I plan to try it someday

**Comment by AkechiShiro** [comment]
Just I've seen `nixos-infect` cannot work on Azure for unknown reasons as of right now, would this be a blocker for this use case ?

**Comment by AkechiShiro** [comment]
I will definitely try and see if we can make this work at some point, not right now tho, but in a month or two

I'm willing to work on It with you

**Comment by KiaraGrouwstra** [comment]
@AkechiShiro you can try opentofu [like this](https://codeberg.org/kiara/teraflops-poc/src/commit/b31e4e93385cd543e97523bca036545de40b5b6e/flake.nix#L18-L38) :)

**Comment by aanderse** [comment]
it would be great if the readme included this - thank you very much for linking this @KiaraGrouwstra!

**Comment by KiaraGrouwstra** [comment]
@AkechiShiro looks like some commands like `set-args` do expect a binary named `terraform` - so i now added an extra package `(pkgs.writeShellScriptBin "terraform" "tofu $@")` to address that.

**Comment by AkechiShiro** [comment]
Thanks a lot for the instructions @KiaraGrouwstra !

**Comment by aanderse** [comment]
> @AkechiShiro looks like some commands like `set-args` do expect a binary named `terraform` - so i now added an extra package `(pkgs.writeShellScriptBin "terraform" "tofu $@")` to address that.

i think it would be nicer if we have the `teraflops` script attempt to use `tofu` if it can't find `terraform`

i will set some time aside soon to document this, etc...

thanks again for your help! ❤️

**Comment by aanderse** [comment]
> EDIT : I think we could close this issue once there is a PR with these instructions in the README @aanderse

@AkechiShiro if you plan on collaborating on `azure` backend we could use this issue as a starting place, otherwise feel free to close the issue if you feel it is no longer relevant 👍

**Comment by AkechiShiro** [comment]
For now I'll be closing it, for the `azure` backend, I may open this issue back later, or open a new one, but at the moment it's not my focus right now.

Thanks for the improvements to use opentofu more easily !
